### PR TITLE
Feature compiler checks

### PR DESCRIPTION
**Pull request by NicoStoll**
Support for:
- WHILE must be closed with WEND
- JUMPS only to existing Line numbers
- VARIABLES must be initialized

### CODE REVIEW

File: src/Lexer.jjt [hunk 1]
```diff
@@ -229,9 +314,12 @@ void ComparisonOperator() #ComparisonOperator : {
 }
 
 void VariableAccess() #VariableAccess : {
+    Token t = new Token();
```
Comment (koebelt) [inline src/Lexer.jjt]: This token is never used so not needed

File: src/Lexer.jjt [hunk 2]
```diff
@@ -244,14 +332,28 @@ void Variable() #Variable : {
...
+        isAssignmentVar = false;
+        jjtThis.jjtSetValue(t.image);
+    }
 }
 
 void Expression() #Expression : {
     Token t = new Token();
 }
 {
-    Term() ( (t=<PLUS> | t=<MINUS>) Term() )* { jjtThis.jjtSetValue(t.image); }
```
Comment (koebelt) [inline src/Lexer.jjt:254]: If you remove `jjtThis.jjtSetValue(t.image);` then no addition or substractions can be recognised,

**BREAKING CHANGE**

**Review by koebelt — CHANGES_REQUESTED** [review]
The Expression `jjtThis.jjtSetValue(t.image);` line 254, now 355, has been deleted and makes +/- operators unusable. Please revert this

**Comment by NicoStoll** [comment]
Removed Token
Added jjtThis.jjtSetValue(t.image)
Added FOR Check

**Review by koebelt — APPROVED** [review]
LGTM, Ready to merge

**Pull request merged**